### PR TITLE
Update base-gas.md: to reflect the removal of the storage curves and …

### PR DESCRIPTION
**Pull request by msmouse**
…introduction of the storage fee

### CODE REVIEW

File: developer-docs-site/docs/concepts/base-gas.md [hunk 1]
```diff
@@ -257,50 +256,20 @@ As of the time of this writing, `min_price_per_gas_unit` in [`transaction.rs`](h
...
 
-Thus, in the absence of a legitimate economic incentive to deallocate from global storage (via either `move_from<T>()` or by removing from a table), the most effective storage gas optimization strategy is as follows:
+On the other hand, releasing state slots by deleting state items are incentivized by the storage refund. For now state slot fee is fully refunded uppon slot deallocation, and the excess state byte fee is not refundable. This is to be changed shortly by distinguishing permanent bytes (those in the global state) and reletive ephemeral bytes (those goes through the ledger history).
 
-1. Minimize per-item creations
-2. Track unused items and overwrite them, rather than creating new items, when possible
-3. Contain per-item writes to as few items as possible
-4. Read, rather than write, whenever possible
-5. Minimize the number of bytes in all operations, especially writes
+1. Minimize state item creation.
```
Comment (sherry-x) [inline developer-docs-site/docs/concepts/base-gas.md:265]: You still want "the most effective storage gas optimization strategy is as follows:" before those numbered items?

File: developer-docs-site/docs/concepts/base-gas.md [hunk 2]
```diff
@@ -190,28 +190,27 @@ Basic instruction gas parameters are defined at [`instr.rs`] and include the fol
 
 Additional storage gas parameters are defined in [`table.rs`], [`move_stdlib.rs`], and other assorted source files in [`aptos-gas-schedule/src/`].
 
-## Storage gas
+## IO and Storage charges
 
-Storage gas is defined in [`storage_gas.move`], which is accompanied by a comprehensive and internally-linked DocGen file at [`storage_gas.md`].
+To reflect cost relavent to transient storage device resources, including disk IOPS and bandwdith, these gas parameters are charged.
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:195]:
```suggestion
The following gas parameters are applied (i.e., charged) to represent the costs associated with transient storage device resources, including disk IOPS and bandwidth:
```

File: developer-docs-site/docs/concepts/base-gas.md [hunk 3]
```diff
@@ -190,28 +190,27 @@ Basic instruction gas parameters are defined at [`instr.rs`] and include the fol
...
 
+| Parameter                       | Meaning                                                            |
+|---------------------------------|--------------------------------------------------------------------|
+| storage_io_per_state_slot_write | charged per state write operation in the transaction output        |
+| storage_io_per_state_byte_write | charged per byte in all state write ops in the transaction output  |
+| storage_io_per_state_slot_read  | charged per item loaded from global state                          |
+| storage_io_per_state_byte_read  | charged per byte loaded from global state                          |
 
-In short:
+To reflect disk space and structural cost on the Aptos authenticated data structure imposed by storing things on the blockchain, including creating things in the global state, emitting events, etc. storage fee is charged, in absolute APT values.
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:204]:
```suggestion
The following storage fee parameters are applied (i.e., charged in absolute APT values) to represent the disk space and structural costs associated with using the [Aptos authenticated data structure](../reference/glossary/#merkle-trees) for storing items on the blockchain. This encompasses actions such as creating things in the global state, emitting events, and similar operations:
```

File: developer-docs-site/docs/concepts/base-gas.md [hunk 4]
```diff
@@ -257,50 +256,22 @@ As of the time of this writing, `min_price_per_gas_unit` in [`transaction.rs`](h
...
+### Storage Fee
 
-* Writes cost the same as reads on a per-item basis: $w_i = r_i$
-* On a per-byte basis, however, writes cost the same as creates: $w_b = c_b$
-* Per-byte writes and creates cost nearly 17 times as much as per-byte reads: $w_b = c_b = 16.\overline{6} r_b$
-* Per-item reads cost 1000 times as much as per-byte reads: $r_i = 1000 r_b$
-* Per-item creates cost 1000 times as much as per-byte creates: $c_i = 1000 c_b$
-* Per-item writes cost 60 times as much as per-byte writes: $w_i = 60 w_b$
+When the network load is low, the gas unit price is expected to be low, as a result all of other aspects of the transaction cost are expected to be cheap. Except for the storage fee, which is priced in terms of absolute APT value. Usually the transaction fee is the dominant aspect of the whole transaction cost, as long as the transaction allocates state slots or write to large state items, emit many or large events, or the transaction itself is large -- all of which occupy disk space on all Aptos nodes, hence charged accordingly.
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:261]:
```suggestion
When the network load is low, the gas unit price is expected to be low, making most aspects of the transaction cost more affordable. However, the storage fee is an exception, as it's priced in terms of absolute APT value. In most instances, the transaction fee is the predominant component of the overall transaction cost. This is especially true when a transaction allocates state slots, writes to sizable state items, emits numerous or large events, or when the transaction itself is a large one. All of these factors consume disk space on Aptos nodes and are charged accordingly.
```

File: developer-docs-site/docs/concepts/base-gas.md [hunk 5]
```diff
@@ -257,50 +256,22 @@ As of the time of this writing, `min_price_per_gas_unit` in [`transaction.rs`](h
...
 
-Hence per-item operations cost 1000 times more than per-byte operations for both reads and creates, but only 60 times more for writes.
+On the other hand, releasing state slots by deleting state items are incentivized by the storage refund. For now state slot fee is fully refunded uppon slot deallocation, and the excess state byte fee is not refundable. This is to be changed shortly by distinguishing permanent bytes (those in the global state) and reletive ephemeral bytes (those goes through the ledger history).
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:263]:
```suggestion
On the other hand, the storage refund incentivizes releasing state slots by deleting state items. The state slot fee is fully refunded upon slot deallocation, while the excess state byte fee is non-refundable. This will soon change by differentiating between permanent bytes (those in the global state) and relative ephemeral bytes (those that traverse the ledger history).
```

File: developer-docs-site/docs/concepts/base-gas.md [hunk 6]
```diff
@@ -257,50 +256,22 @@ As of the time of this writing, `min_price_per_gas_unit` in [`transaction.rs`](h
...
 
-Thus, in the absence of a legitimate economic incentive to deallocate from global storage (via either `move_from<T>()` or by removing from a table), the most effective storage gas optimization strategy is as follows:
+Some cost optimization strategy with regard to storage fee:
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:265]:
```suggestion
Some cost optimization strategies concerning the storage fee:
```

File: developer-docs-site/docs/concepts/base-gas.md [hunk 7]
```diff
@@ -257,50 +256,22 @@ As of the time of this writing, `min_price_per_gas_unit` in [`transaction.rs`](h
...
-2. Track unused items and overwrite them, rather than creating new items, when possible
-3. Contain per-item writes to as few items as possible
-4. Read, rather than write, whenever possible
-5. Minimize the number of bytes in all operations, especially writes
+1. Minimize state item creation.
+2. Minimize event emission.
+3. Avoid large state items, events and transactions.
+4. Cleaning up state items when no longer used.
+5. If two fields are always updated together, group them into the same resoruce or resource group.
+6. If a struct is large and only a few of the fields get updated frequently, separate out the fields to a separate resource / resource group.
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:272]:
```suggestion
2. Minimize event emissions.
3. Avoid large state items, events, and transactions.
4. Clean up state items that are no longer in use.
5. If two fields are consistently updated together, group them into the same resource or resource group.
6. If a struct is large and only a few fields are updated frequently, move those fields to a separate resource or resource group.
```

**Comment by msmouse** [comment]
Thank you so much @sherry-x  &  @saharct

File: developer-docs-site/docs/concepts/base-gas.md [hunk 8]
```diff
@@ -190,28 +190,27 @@ Basic instruction gas parameters are defined at [`instr.rs`] and include the fol
...
 
+| Parameter                       | Meaning                                                            |
+|---------------------------------|--------------------------------------------------------------------|
+| storage_io_per_state_slot_write | charged per state write operation in the transaction output        |
+| storage_io_per_state_byte_write | charged per byte in all state write ops in the transaction output  |
+| storage_io_per_state_slot_read  | charged per item loaded from global state                          |
+| storage_io_per_state_byte_read  | charged per byte loaded from global state                          |
 
-In short:
+The following storage fee parameters are applied (i.e., charged in absolute APT values) to represent the disk space and structural costs associated with using the [Aptos authenticated data structure](../reference/glossary/#merkle-trees) for storing items on the blockchain. This encompasses actions such as creating things in the global state, emitting events, and similar operations:
```
Comment (saharct) [inline developer-docs-site/docs/concepts/base-gas.md:204]:
```suggestion
The following storage fee parameters are applied (i.e., charged in absolute APT values) to represent the disk space and structural costs associated with using the [Aptos authenticated data structure](../reference/glossary.md#merkle-trees) for storing items on the blockchain. This encompasses actions such as creating things in the global state, emitting events, and similar operations:
```

**Review by saharct — APPROVED** [review]

**Comment by msmouse** [comment]
Will land after storage refund is enabled.

**Review by grao1991 — APPROVED** [review]
